src/holon/services/query_and_convert.py

```diff
@@ -315,23 +322,53 @@ def set_datamodel(self, c: DatamodelConversion):
-        unique_actors_group = self.copied_scenario.actor_set.distinct("group")
-
-        for actor in unique_actors_subgroup:
-            g_value = qr.get_filter_aggregation_result(
-                self.copied_scenario, group_to_filter=actor.group, subgroup_to_filter=actor.subgroup
+        if self.config.config_db.module == QueryCovertModuleType.COSTBENEFIT:
+            actor_repository = self.scenario_aggregate.get_repository_for_model_type(
+                ModelType.ACTOR
             )
-            keyname = CostItem.subgroup(actor)
-            self.distribution_key.update({keyname: g_value / value if value != 0 else 0})
-
-        for actor in unique_actors_group:
-            g_value = qr.get_filter_aggregation_result(
-                self.copied_scenario, group_to_filter=actor.group
+            actor_group_repository = self.scenario_aggregate.get_repository_for_model_type(
+                ModelType.ACTOR_GROUP
             )
-            keyname = CostItem.group(actor)
-            self.distribution_key.update({keyname: g_value / value if value != 0 else 0})
+            actor_sub_group_repository = self.scenario_aggregate.get_repository_for_model_type(
+                ModelType.ACTOR_SUB_GROUP
+            )
+
+            unique_sub_group_ids = actor_repository.distinct(["group_id", "subgroup_id"])
+            unique_sub_groups = {
+                # TODO null groups are not handled here
+                (
+                    actor_group_repository.get(d["group_id"])
+                    if d["group_id"] is not None
+                    else None,
+                    actor_sub_group_repository.get(d["subgroup_id"])
+                    if d["subgroup_id"] is not None
+                    else None,
+                )
+                for d in unique_sub_group_ids
+            }
+
+            for group_tuple in unique_sub_groups:
+                g_value = qr.get_filter_aggregation_result(
+                    scenario_aggregate=self.scenario_aggregate,
+                    group_to_filter=group_tuple[0],
+                    subgroup_to_filter=group_tuple[1],
+                )
+                # TODO null groups are not handled here
+                keyname = CostItem.group_key_name(group_tuple[0], group_tuple[1])
+                self.distribution_key.update({keyname: g_value / value if value != 0 else 0})
+
+            # TODO: aren't we double counting now? some guys already got some in their subgroup?
+            unique_group_ids = actor_repository.distinct(["group_id"])
+            # TODO null groups are not handled here
+            unique_groups = [actor_group_repository.get(d["group_id"]) for d in unique_group_ids]
+
+            for group in unique_groups:
+                g_value = qr.get_filter_aggregation_result(
+                    scenario_aggregate=self.scenario_aggregate, group_to_filter=group
+                )
+                # TODO null groups are not handled here
```

Comment:
Zijn dit TODOs die nog in deze PR opgepakt moet worden?

Reply:
Deze had ik wel al gedaan inmiddels.